share/getters/shrex.go

```diff
@@ -34,6 +34,11 @@ func (sg *ShrexGetter) GetShare(ctx context.Context, root *share.Root, row, col
 
 func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.ExtendedDataSquare, error) {
 	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
```

Comment:
alternative and shorter form
```suggestion
	for ctx.Err == nil {
```

Reply:
based

Reply:
we use select pattern everywhere else so lets stick to it but im fine w shorter form if we unify later on